mesonbuild/msubprojects.py

```diff
@@ -435,6 +436,33 @@ def update(self) -> bool:
             self.wrap.update_hash_cache(self.repo_dir)
         return success
 
+    def syncwrap(self) -> bool:
+        if self.wrap.type != 'git' and not self.wrap.redirected:
+            return True
+
+        self.log(f'Sync {self.wrap.name}...')
+        if self.wrap.redirected:
+            self.wrap.filename = Path(self.wrap.original_filename).resolve()
+            self.wrap.parse_wrap()
+
+        if self.wrap.type != 'git':
+            return True
+
+        revision = self.wrap.get('revision').strip()
+        if revision.lower() == 'head':
+            return True
+
+        new_revision = self.git_output(['rev-parse', 'HEAD']).strip()
+        if new_revision != revision:
+            if self.wrap.redirected:
+                self.log(mlog.yellow('Warning:'), f'{self.wrap.original_filename} redirected to {self.wrap.filename}')
+            cp = configparser.ConfigParser(interpolation=None)
+            cp.read(self.wrap.filename)
+            cp['wrap-git']['revision'] = new_revision
+            with open(self.wrap.filename, 'w', encoding='utf-8') as f:
+                cp.write(f)
```

Comment:
For consistency with other commands, better add something like `self.log('  -> New revision {new_revision}')`.

Reply:
Will do.